ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -7949,7 +7949,15 @@ protected Operator genFileSinkPlan(String dest, QB qb, Operator input)
           }
         } else {
           if (tblDesc.isCTAS() && tblDesc.getStorageHandler() != null) {
-            tblDesc.setLocation(getCtasOrCMVLocation(tblDesc, viewDesc, createTableUseSuffix).toString());
+            try {
```

Comment:
can't we simply do `if (tblDesc.isCTAS() && isIcebergTable(tblProps)) {`

Reply:
btw instead `createTableUseSuffix` better to directly use `false`

Reply:
This part of the code is in the compiler and it should not depend on Iceberg or any storage format directly. Hive has `HiveIcebergStorageHandler` interface to hide storage format customizations.

Reply:
"btw instead createTableUseSuffix better to directly use false"
changed to false

Reply:
I agree that compiler code should be `storage format` agnostic, however:
1. SA already has specialized logic for external / ACID / Iceberg (`isIcebergTable()`) --> requires refactor, we can ignore for now;
2. `HiveStorageHandler` is being abused and polluted with many meaningless methods. `SH#requiresLocation()` - what does it even mean? Could we please use self-explanatory naming in the interface?